sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -42,6 +42,25 @@ options { tokenVocab = SqlBaseLexer; }
   public boolean double_quoted_identifiers = false;
 }
 
+batchOrSingleStatement
+    : batchCompound SEMICOLON* EOF
```

Comment:
We should figure out the naming.

We should be clear about what is `batch` and what is a `compound`. At this moment line is blurry.

Also, we should follow same rules as singleStatement - let's avoid EOF check here, but add something like `singleBatch` which will contain EOF check at the end.

Reply:
Agreed offline on the naming.